SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/DefaultPipeline.cpp

```diff
@@ -90,19 +92,23 @@ void DefaultPipeline::doCollisionReset()
     msg_info_when(d_doPrintInfoMessage.getValue())
             << "DefaultPipeline::doCollisionReset" ;
 
-    core::objectmodel::BaseContext* scene = getContext();
     // clear all contacts
     if (contactManager!=nullptr)
     {
         const helper::vector<Contact::SPtr>& contacts = contactManager->getContacts();
-        for (helper::vector<Contact::SPtr>::const_iterator it = contacts.begin(); it!=contacts.end(); ++it)
+        for (const auto& contact : contacts)
         {
-            (*it)->removeResponse();
+            if (contact != nullptr)
```

Comment:
The `DefaultContactManager` makes sure there is no null contacts when creating them (phew). Does it mean I can trust him? No. What about `RuleBasedContactManager`? I don't know this component. There is no contract visible to `DefaultPipeline` so a ContactManager keeps only non-null contacts. It is also possible they have been nullified since the contact manager created them.
It's always a good practice to check nullptr. We can never anticipate that a pointer is always valid.